src/test/rustdoc/deref-recursive-cycle.rs

```diff
@@ -0,0 +1,16 @@
+// #26207: Ensure `Deref` cycles are properly handled without errors.
```

Comment:
nit: I would put this in `rustdoc-ui` with `// check-pass` to make clear we're just testing that rustdoc doesn't crash.

Reply:
Thanks, I moved the test as you described. 😄